server/aesdsocket.c

```diff
@@ -92,227 +114,441 @@ void signal_handler(int signal_number) {
-    error_handler(Socket_created);
-    exit(EXIT_FAILURE);
+  file_fd = open(DATA_FILE, O_CREAT | O_RDWR | O_APPEND, 0744);
+  if (file_fd == -1) {
+    syslog(LOG_ERR,
+           "ERROR: Failed to  OPEN the FILE at /var/temp/aesdsocketdata");
+    error_handler(thread_close);
+    node->my_thread_complete = false;
+    return thread_node;
+    // exit(EXIT_FAILURE);
   }
 
-  if ((argc == 2) && (strcmp(argv[1], "-d") == 0)) {
-    pid_t pid;
+  while (1) {
+    // bzero(buf, BUF_SIZE);
+    memset(buf, 0, BUF_SIZE);
+    int n_received = recv(node->client_socket_fd, buf, 1024, 0);
+    // printf("\n\r data received is %s\n\r",buf);
+    if (n_received == -1) {
+      syslog(LOG_ERR, "ERROR: Failed to RECEIVE data");
+      error_handler(thread_close);
+      node->my_thread_complete = false;
+      return thread_node;
+      // exit(EXIT_FAILURE);
+    }
+
+    if (pthread_mutex_lock(node->thread_mutex) != 0) {
+      syslog(LOG_ERR, "ERROR: Failed to lock mutex");
+      error_handler(thread_close);
+      node->my_thread_complete = false;
+      return thread_node;
+    }
+
+    int n_written = write(file_fd, buf, n_received);
+    //printf("\n\r data writing to file  is %s\n\r", buf);
+    if ((n_written != n_received)) {
+      syslog(LOG_ERR, "ERROR: Failed to WRITE complete data");
+      error_handler(thread_close);
+      pthread_mutex_unlock(node->thread_mutex);
+      node->my_thread_complete = false;
+      return thread_node;
+      // exit(EXIT_FAILURE);
+    }
+
+    if (pthread_mutex_unlock(node->thread_mutex) != 0) {
+      syslog(LOG_ERR, "ERROR: Failed to unlock mutex");
+      error_handler(thread_close);
+      node->my_thread_complete = false;
+      return thread_node;
+    }
+
     int i = 0;
+    // or can you some standerd functions like memchr
+    for (i = 0; i < n_received; i++) {
+      if (buf[i] == '\n') {
+        packet_receive_complete = 1;
+        break;
+      }
+    }
 
-    /* create new process If parent, log and exit successfully*/
-    pid = fork();
-    if (pid == -1) {
-      syslog(LOG_ERR, "ERROR: FORK failed");
-      exit(EXIT_FAILURE);
-    } else if (pid != 0) {
-      syslog(LOG_INFO, "INFO- Parent");
-      exit(EXIT_SUCCESS);
+    // If newline found, break out of the loop
+    if (packet_receive_complete == 1) {
+      break;
     }
-    // Child process continues here
-    syslog(LOG_INFO, "INFO- Child");
-    // Create a new session for the child process
-    if (setsid() == -1) {
-      syslog(LOG_ERR, "ERROR: SESSION FAIL");
-      exit(EXIT_FAILURE);
+  }
+  struct stat file_info;
+  fstat(file_fd, &file_info);
+  int file_size = file_info.st_size;
+  int cursor_set = lseek(file_fd, 0, SEEK_SET);
+  if (cursor_set == -1) {
+    syslog(LOG_ERR, "ERROR: Failed to SEEK cursor to start");
+    error_handler(thread_close);
+    node->my_thread_complete = false;
+    return thread_node;
+    // exit(EXIT_FAILURE);
+  }
+  int byte_transfer_size = 1024;
```

Comment:
@aneesh1298 why the hard-coded 1024 everywhere?  This should be a const int or similar.